state/redis/redis.go

```diff
@@ -245,19 +246,20 @@ func (r *StateStore) BulkSet(req []state.SetRequest) error {
 
 // Multi performs a transactional operation. succeeds only if all operations succeed, and fails if one or more operations fail
 func (r *StateStore) Multi(operations []state.TransactionalRequest) error {
-	redisReqs := []redis.Request{}
+
+	pipe := r.client.TxPipeline()
 	for _, o := range operations {
 		if o.Operation == state.Upsert {
 			req := o.Request.(state.SetRequest)
 			b, _ := r.json.Marshal(req.Value)
-			redisReqs = append(redisReqs, redis.Req("SET", req.Key, b))
+			pipe.Set(req.Key, b, 0)
```

Comment:
Please avoid hardcoding, make it as defaultExpirationTime =0 and use it through variable .
Also consider reading the key expiration property from the request options, if it is not there use default..

Reply:
(y) Added the constant now.

> Also consider reading the key expiration property from the request options, if it is not there use default..

I assume we need to add that to state.SetRequest struct first?

Reply:
Lets not change `state.SetRequest` at this time.

Reply:
@pruthvidhodda I would suggest to get the existing functionality working with TLS and new redis package. More configs can be added later.